### PR TITLE
OADP-849: Fix race conditions in VSR controller

### PR DESCRIPTION
**Pull request by eemcmullan**

### CODE REVIEW

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by weshayutin — APPROVED** [review]
Thank you!!

File: controllers/replicationdestination.go [hunk 1]
```diff
@@ -42,10 +42,21 @@ func (r *VolumeSnapshotRestoreReconciler) CreateReplicationDestination(log logr.
...
+		}
+		r.Log.Error(err, fmt.Sprintf("unable to fetch restic secret %s/%s", r.NamespacedName.Namespace, dmresticSecretName))
+		return false, err
+	}
+
 	// Create ReplicationDestination in protected namespace
 	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, repDestination, func() error {
 
-		return r.buildReplicationDestination(repDestination, &vsr)
+		return r.buildReplicationDestination(repDestination, &vsr, &resticSecret)
```
Comment (kaovilai) [inline controllers/replicationdestination.go:59]: Restic secret isn't being used here. Are we planning to use it in the future?

Comment (eemcmullan) [reply controllers/replicationdestination.go:59]: It's being passed to replicationDestination here to add to its spec

Comment (kaovilai) [reply controllers/replicationdestination.go:59]: I see now.. it wasn't part of the diff so I missed it.. thanks for the explanation!

**Pull request merged**